crates/biome_js_analyze/src/semantic_analyzers/nursery/no_invalid_new_builtin.rs

```diff
@@ -59,38 +87,40 @@ impl Rule for NoInvalidNewBuiltin {
         let callee = ctx.query().callee().ok()?;
         let (reference, name) = global_identifier(&callee)?;
         match name.text() {
-            "Symbol" | "BigInt" => ctx.model().binding(&reference).is_none().then_some(name),
+            "Symbol" | "BigInt" | "Math" | "JSON" | "Reflect" | "Atomics" | "Intl" => {
+                ctx.model().binding(&reference).is_none().then_some(name)
+            }
             _ => None,
         }
     }
 
-    fn diagnostic(
-        ctx: &RuleContext<Self>,
-        builtin_fn_name: &Self::State,
-    ) -> Option<RuleDiagnostic> {
+    fn diagnostic(ctx: &RuleContext<Self>, builtin_name: &Self::State) -> Option<RuleDiagnostic> {
         Some(RuleDiagnostic::new(
             rule_category!(),
             ctx.query().range(),
             markup! {
-                <Emphasis>"`"{builtin_fn_name.text()}"`"</Emphasis>" cannot be called as a constructor."
+                <Emphasis>"`"{builtin_name.text()}"`"</Emphasis>" cannot be called as a constructor."
             },
         ))
     }
 
-    fn action(ctx: &RuleContext<Self>, _: &Self::State) -> Option<JsRuleAction> {
+    fn action(ctx: &RuleContext<Self>, builtin_name: &Self::State) -> Option<JsRuleAction> {
         let node = ctx.query();
         let call_expression = convert_new_expression_to_call_expression(node)?;
         let mut mutation = ctx.root().begin();
         mutation.replace_node_discard_trivia::<AnyJsExpression>(
             node.clone().into(),
             call_expression.into(),
         );
-        Some(JsRuleAction {
-            category: ActionCategory::QuickFix,
-            applicability: Applicability::MaybeIncorrect,
-            message: markup! { "Remove "<Emphasis>"`new`"</Emphasis>"." }.to_owned(),
-            mutation,
-        })
+        match builtin_name.text() {
+            "Symbol" | "BigInt" => Some(JsRuleAction {
+                category: ActionCategory::QuickFix,
+                applicability: Applicability::MaybeIncorrect,
+                message: markup! { "Remove "<Emphasis>"`new`"</Emphasis>"." }.to_owned(),
+                mutation,
+            }),
+            _ => None,
+        }
```

Comment:
I don't see how this rule can provide code fix actions for built-in objects, so they are ignored in `action`.